python_src/src/lamp_py/performance_manager/flat_file.py

```diff
@@ -18,63 +26,181 @@
+    vehicle_events_service_dates = db_manager.select_as_dataframe(
+        sa.select(VehicleEvents.service_date).distinct()
+    )
+    processed_datetimes = db_service_dates_to_datetimes(
+        vehicle_events_service_dates
+    )
+
+    # get service dates with new data from the last event loop
+    new_data_service_dates = db_manager.select_as_dataframe(
+        sa.select(TempEventCompare.service_date).distinct()
+    )
+    new_data_datetimes = db_service_dates_to_datetimes(new_data_service_dates)
+
+    # return the processed dates that have yet to be archived plus dates with new data
+    return (processed_datetimes - archived_datetimes).union(new_data_datetimes)
+
+
 def write_flat_files(db_manager: DatabaseManager) -> None:
-    """write flat files to s3 for datetimes"""
+    """
+    * find service dates that have not been fully archived
+    * write flat files for those dates
+    * update the archive log csv file
+    """
     # if we don't have a public archive bucket, exit
-    if os.environ.get("PUBLIC_ARCHIVE_BUCKET", "") == "":
+    if S3Archive.BUCKET_NAME == "":
         return
 
-    date_df = db_manager.select_as_dataframe(
-        sa.select(TempEventCompare.service_date).distinct()
-    )
+    service_dates = dates_to_update(db_manager)
 
-    # if no data has been processed, exit
-    if date_df.shape[0] == 0:
+    # if no data to archive, early exit
+    if len(service_dates) == 0:
         return
 
     process_logger = ProcessLogger(
-        "bulk_flat_file_write", date_count=date_df.shape[0]
+        "bulk_flat_file_write", date_count=len(service_dates)
     )
     process_logger.log_start()
 
     s3_directory = os.path.join(
-        os.environ["PUBLIC_ARCHIVE_BUCKET"], "lamp", "flat_file"
+        S3Archive.BUCKET_NAME, S3Archive.RAIL_PERFORMANCE_PREFIX
     )
 
-    for date in date_df["service_date"]:
-        sub_process_logger = ProcessLogger("flat_file_write", service_date=date)
+    for service_date in service_dates:
+        sub_process_logger = ProcessLogger(
+            "flat_file_write", service_date=service_date
+        )
         sub_process_logger.log_start()
 
         try:
-            as_str = str(date)
-            filename = f"{as_str[0:4]}-{as_str[4:6]}-{as_str[6:8]}-rail-performance-{{i}}.parquet"
+            date_str = service_date.strftime("%Y-%m-%d")
+            filename = f"{date_str}-subway-on-time-performance-v1.parquet"
 
-            flat_table = generate_daily_table(db_manager, date)
+            flat_table = generate_daily_table(db_manager, service_date)
             sub_process_logger.add_metadata(row_count=flat_table.shape[0])
 
             write_parquet_file(
                 table=flat_table,
                 file_type="flat_rail_performance",
                 s3_dir=s3_directory,
-                partition_cols=["year", "month", "day"],
-                basename_template=filename,
+                filename=filename,
             )
```

Comment:
You could use our new `DatabaseManager.write_to_parquet` functionality here, to avoid all the DB -> pandas -> pyarrow table -> parquet logic.

Reply:
done and done.